File: TwinTrades/TwinTrades/Models/Subscription.swift
import Foundation

enum SubscriptionPlan: String, Codable {
    case monthly
    case annual
    case lifetime
}

enum SubscriptionStatus: String, Codable {
    case active
    case expired
    case cancelled
    case grace_period
}

struct Subscription: Codable, Identifiable {
    let id: Int
    let plan: SubscriptionPlan
    let status: SubscriptionStatus
    let appleTransactionId: String
    let amountPaid: Decimal
    let currency: String
    let expiresAt: Date?
    let createdAt: Date

    enum CodingKeys: String, CodingKey {
        case id, plan, status, currency
        case appleTransactionId = "apple_transaction_id"
        case amountPaid = "amount_paid"
        case expiresAt = "expires_at"
        case createdAt = "created_at"
    }
}

struct VerifyApplePayload: Encodable {
    let transactionId: String
    let productId: String

    enum CodingKeys: String, CodingKey {
        case transactionId = "transaction_id"
        case productId = "product_id"
    }
}

File: frontend/models/Subscription.ts
export type SubscriptionPlan = 'monthly' | 'annual' | 'lifetime';
export type SubscriptionStatus = 'active' | 'expired' | 'cancelled' | 'grace_period';

export interface Subscription {
  id: number;
  plan: SubscriptionPlan;
  status: SubscriptionStatus;
  apple_transaction_id: string;
  amount_paid: string;
  currency: string;
  expires_at: string | null;
  created_at: string;
}

export interface VerifyApplePayload {
  transaction_id: string;
  product_id: string;
}

export type BillingEventType =
  | 'payment_success'
  | 'payment_failed'
  | 'renewal'
  | 'refund'
  | 'cancellation';

export interface BillingEvent {
  id: number;
  event_type: BillingEventType;
  amount: string;
  currency: string;
  apple_transaction_id: string | null;
  occurred_at: string;
}

export interface PaginatedBillingHistoryResponse {
  events: BillingEvent[];
  total: number;
  page: number;
  page_size: number;
}
